jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/ClientUpgradeRequest.java

```diff
@@ -297,6 +292,25 @@ public void setSession(Object session)
         throw new UnsupportedOperationException("HttpSession not available on Client request");
     }
 
+    /**
+     * @param timeout the total timeout for the request/response conversation of the WebSocket handshake;
+     * use zero or a negative value to disable the timeout
+     * @param unit the timeout unit
+     */
+    public void setTimeout(long timeout, TimeUnit unit)
+    {
+        this.timeout = unit.toMillis(timeout);
+    }
+
+    /**
+     * @return the total timeout for this request, in milliseconds;
+     * zero or negative if the timeout is disabled
+     */
+    public long getTimeout()
+    {
+        return timeout;
+    }
+
     /**
      * ABNF from RFC 2616, RFC 822, and RFC 6455 specified characters requiring quoting.
      */
```

Comment:
Please move this static field to the top of the class and make it `private`.

Also, no need to pass it as parameter to `quoteIfNeeded()`.

Reply:
This suggestion is unrelated to this issue.
Good for a different PR related to general cleanup tho.

Reply:
I will do this in a separate cleanup PR then.